### ISSUE TITLE
January Commit to camkes-vm-linux breaks camkes-vm-crossvm Tutorial

### ISSUE THREAD

**Issue by jimshift5** [seL4/sel4-tutorials#49]
This commit appears to remove the libraries necessary to complete the crossvm tutorial: https://github.com/SEL4PROJ/camkes-vm-linux/commit/bab3e9c2b52ff942b6eead71e33ad781b429a646.

Doing a git hard reset also does not solve the problem as camkes_mutex.h and a few others are also missing. Is there an updated tutorial somewhere since the point of this commit appears to depreciate the old crossvm connector modules?

**Comment by jimshift5** [comment]
I just noticed under SEL4PROJ/camkes-arm-vm there is an example app for vm_cross_connector (for anyone with the same question it is here: https://github.com/SEL4PROJ/camkes-arm-vm) so I will use that as an exemplar. Therefore please disregard the second part of the issue.

**Comment by chuahou** [comment]
I think this issue should remain open since it is still a problem. Currently, following instructions on https://docs.sel4.systems/Tutorials/camkes-vm-crossvm.html still has this problem. Running, for example,

```sh
./init --tut camkes-vm-crossvm --solution
ninja -C camkes-vm-crossvm_build
```

complains about the missing `camkes-connector-apps/libs/camkes` removed in the commit you mentioned.

It would probably be best to update the tutorial, though I lack the requisite knowledge to do that (which is why I'm doing the tutorial now). For now it seems sufficient to work through the examples on [seL4/camkes-vm-examples](https://github.com/seL4/camkes-vm-examples) instead.

**Comment by lsf37** [comment]
You're right, if the tutorial as described is broken, this should definitely remain open.

**Comment by jasonleecode** [comment]
this issue exist and not fixed so far.
